### PR TITLE
adding new validation for future dates to admin in customer DOB - issue #22692

### PR DESCRIPTION
**Pull request by tiagooctp**
### Description (*)
We have tried to create new customer from Magento back end and added Future date as Date of Birth example - 09/09/2020 and we can save customer without any issue

### Fixed Issues
On Admin, when creating a new customer, there was no validation for future dates. It was possible to add dates greater than the current date. 
Added new validation to the field.

1. magento/magento2#22692: Add New Customer from admin panel - DOB field validation missing for future dates

### Manual testing scenarios (*)
1. Log in to Admin > Customer > All customers > Add new customer.
2. Change the DoB date for invalid dates or greater than current date.
3. Try to save new customer

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by rodrigowebjump** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-24588.instances.magento-community.engineering).
Admin access: [https://pr-24588.instances.magento-community.engineering/admin](https://pr-24588.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-24588.instances.magento-community.engineering).
Admin access: [https://pr-24588.instances.magento-community.engineering/admin](https://pr-24588.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-24588.instances.magento-community.engineering).
Admin access: [https://pr-24588.instances.magento-community.engineering/admin](https://pr-24588.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-24588.instances.magento-community.engineering).
Admin access: [https://pr-24588.instances.magento-community.engineering/admin](https://pr-24588.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rodrigowebjump** [comment]
@magento #mm19br

**Review by rodrigowebjump — APPROVED** [review]
Hi @tiagooctp Thanks for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, thank you for the review.
[ENGCOM-5839](https://jira.corp.magento.com/browse/ENGCOM-5839) has been created to process this Pull Request
:eight_spoked_asterisk: @rodrigowebjump, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @rodrigowebjump , could you, please, review PR with last commit? Thanks!

**Comment by tiagooctp** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @tiagooctp. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @tiagooctp, here is your [new Magento instance](https://pr-24588.instances.magento-community.engineering).
Admin access: [https://pr-24588.instances.magento-community.engineering/admin](https://pr-24588.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5839](https://jira.corp.magento.com/browse/ENGCOM-5839) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
Hi @tiagooctp 
Could you, please, add this phrase `The Date of Birth should not be greater than today.` to translation file?
Thanks!

File: app/code/Magento/Customer/view/frontend/web/js/validation.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require([
```
Comment (DrewML) [inline app/code/Magento/Customer/view/frontend/web/js/validation.js]: This should be in a `define` wrapper.

```define(['jquery', 'moment', 'jquery/validate', 'mage/translate'], function() {
   // module body
});

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js [hunk 2]
```diff
@@ -1067,6 +1067,16 @@ define([
                 return new RegExp(param).test(value);
             },
             $.mage.__('This link is not allowed.')
+        ],
```
Comment (DrewML) [inline app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js:1070]: I'm not super familiar with this section of the code - what's the reason we have the same implementation in `validation/rules.js` and `js/validation.js`?

Comment (tiagooctp) [reply app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js:1070]: Hi @DrewML 
The validation/rules.js is only used on the admin aprt. I've created the js/validation.js so I could add the validation on the frontend also.

File: app/code/Magento/Customer/view/frontend/web/js/validation.js (hunk 1 above)
Comment (tiagooctp) [reply app/code/Magento/Customer/view/frontend/web/js/validation.js]: @DrewML I will change it asap

Comment (DrewML) [reply app/code/Magento/Customer/view/frontend/web/js/validation.js]: Thanks! Appreciate it

**Comment by sidolov** [comment]
@tiagooctp  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by tiagooctp** [comment]
Hi @sidolov 
I've reopenned and fixed the issue. 
I've already signed the CLA but the agreements do not update.

**Review by rodrigowebjump — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, thank you for the review.
[ENGCOM-6057](https://jira.corp.magento.com/browse/ENGCOM-6057) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**